### PR TITLE
ARROW-18237: [Java] Extend Table code

### PR DESCRIPTION
**Pull request by lwhite1**
Initial merge of MutableTable / MutableRow plus tests.

### CODE REVIEW

**Comment by lwhite1** [comment]
@davisusanibar Would you please review.  This is a partial implementation but it's already quite large.  The code is mostly very simple and well tested. I think before adding anything more it would be best to review and merge what's here

**Review by davisusanibar — APPROVED** [review]
LGTM, thanks for all the documentation added on the methods/classes (+TODO).

**Comment by lwhite1** [comment]
@lidavidm Could you please give this a final check and merge? It was originally intended to include new code for mutable table support but has been simplified to include adding a small number of necessary methods to the existing classes.

File: java/vector/src/main/java/org/apache/arrow/vector/table/Row.java [hunk 1]
```diff
@@ -98,6 +102,8 @@
 import org.apache.arrow.vector.holders.NullableUInt8Holder;
 
 /**
+ * TODO: Modify the getters for Duration and others so that they return something better than ArrowBuf when possible
```
Comment (lidavidm) [inline java/vector/src/main/java/org/apache/arrow/vector/table/Row.java:105]: Did you mean to leave this TODO in here? (It should perhaps be linked to an issue?)

**Review by lidavidm — APPROVED** [review]

File: java/vector/src/main/java/org/apache/arrow/vector/table/Row.java (hunk 1 above)
Comment (lwhite1) [reply java/vector/src/main/java/org/apache/arrow/vector/table/Row.java]: removed obsolete TODOs

**Review by lidavidm — APPROVED** [review]

**Pull request merged**